test/gtest/graphapi_mha_bwd.cpp

```diff
@@ -0,0 +1,371 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "graphapi_mha_cpp_common.hpp"
+
+namespace mha_graph_test {
+
+class MhaBwdGraphTest : public MhaGraphTestBase
+{
+
+    tensor<float> mSoftMax;
+
+public:
+    void createMhaGraph(size_t n, size_t h, size_t s, size_t d) override
+    {
+        mGraphBuilder = std::make_unique<gr::OpGraphBuilder>();
+
+        std::vector<int64_t> nhsd  = {n, h, s, d};
+        std::vector<int64_t> nhss  = {n, h, s, s};
+        std::vector<int64_t> nhs1  = {n, h, s, 1};
+        std::vector<int64_t> all1s = {1, 1, 1, 1};
+
+        MAKE_TENSOR_F(Q, nhsd, false);
+        MAKE_TENSOR_F(K, nhsd, false);
+        MAKE_TENSOR_F(V, nhsd, false);
+        MAKE_TENSOR_F(dO, nhsd, false);
+        MAKE_TENSOR_F(O, nhsd, false);
+
+        MAKE_TENSOR_F(T_MM_0, nhss, true);
```

Comment:
[R] If we can't live without that wrapper, I'd rather moved it to something like `graphapi_common.hpp` or stick to the raw API.
Seems it's kind of widely used case.

Reply:
It saves typing in these tests, so I will move it up in the future when the need arises.